### ISSUE TITLE
Rule proposal: prefer `declare var` for global declarations

### ISSUE THREAD

**Issue by ark120202** [typescript-eslint/typescript-eslint#2594]
Using block-scoped `declare const` and `declare let` in global context has a potentially unwanted effect - they can be accessed using unqualified access, but not using `globalThis`. When this behavior have been implemented in TypeScript (https://github.com/microsoft/TypeScript/pull/30510) all declarations in standard lib files have been changed to `declare var`, so I believe it's something that generally can be recommended for external declarations too.

```ts
// Invalid
declare const foo: string;
```

```ts
// Invalid
export {}
declare global {
	const foo: string;
}
```

```ts
// Valid
export {}
declare const foo: string;
```

**Comment by kirkwaiblinger** [comment]
Note that the "accepting prs" should be taken with a grain of salt here while #7941 is being discussed

**Comment by bradzacher** [comment]
Based on our current expectations for enhancement requests, I believe this request fails the "nicheness" test:
- It specifically applies to authoring `.d.ts` files -- which few people actively do -- the vast, vast majority of people don't write (or really even read) them. This makes it a very, very niche proposal
- This request has been open for 6 years and has netted a total of non-team member 3 interactions.
- This request has been open for 6 years and nobody has moved to implement it.

For these reasons, I believe we should move to reject and close this. Thoughts @typescript-eslint/triage-team?

**Comment by kirkwaiblinger** [comment]
I agree.

While I still think the idea has value, I think it would fit better as a proposal to ESLint core nowadays anyways rather than having to be an extension rule in typescript-eslint
